app/docs/0.12.x/healthchecks.md

```diff
@@ -0,0 +1,182 @@
+        "active": {
+            "concurrency": 10,
+            "healthy": {
+                "http_statuses": [ 200, 302 ],
+                "interval": 0,
+                "successes": 0
+            },
+            "http_path": "/",
+            "timeout": 1,
+            "unhealthy": {
+                "http_failures": 0,
+                "http_statuses": [ 429, 404, 500, 501,
+                                   502, 503, 504, 505 ],
+                "interval": 0,
+                "tcp_failures": 0,
+                "timeouts": 0
+            }
+        },
+        "passive": {
+            "healthy": {
+                "http_statuses": [ 200, 201, 202, 203,
+                                   204, 205, 206, 207,
+                                   208, 226, 300, 301,
+                                   302, 303, 304, 305,
+                                   306, 307, 308 ],
+                "successes": 0
+            },
+            "unhealthy": {
+                "http_failures": 0,
+                "http_statuses": [ 429, 500, 503 ],
+                "tcp_failures": 0,
+                "timeouts": 0
+            }
+        }
+    },
+    "slots": 10
+}
+```
+
+If all targets of an upstream are unhealthy, Kong will respond to requests
+to the upstream with `503 Service Unavailable`.
+
+Note that health checks operate only on ["active" targets][4] and do not
+modify the "active" status of a target in the Kong database.
+
+[4]: /docs/{{page.kong_version}}/admin-api#target-object
+
+### Active health checks
+
+**Active health checks**, as the name implies, actively probe targets for
+their health. When a target becomes unresponsive, the active health
+checker will detect that, mark the node as unhealthy, and keep checking
+it periodically, automatically marking the node as healthy once
+it becomes responsive again.
+
+When active health checks are enabled in an upstream entity, Kong will
+periodically issue HTTP requests to a configured path at each target of
+the upstream.
+
+The periodicity of active health checks can be configured separately for
+when a target is healthy or unhealthy. If the `interval` value for either
+is set to zero, the checking is disabled at the corresponding scenario.
+When both are zero, active health checks are disabled altogether.
+
+[Back to TOC](#table-of-contents)
+
+### Passive health checks (circuit breakers)
+
+**Passive health checks**, also known as **circuit breakers**, are
+checks performed based on the data being regularly proxied by Kong only,
+with no additional traffic being generated. When a target becomes
+unresponsive, the passive health checker will detect that and mark
+the node as unhealthy. The ring-balancer will start skipping this 
+target, so no more traffic will be routed to it. 
+
+Passive health checks have the advantage of not producing extra
+traffic, but they are unable to automatically mark a node as 
+healthy again: the "circuit is broken", and the target needs to
```

Comment:
"healthy again - once the circuit is broken, the target will need to be re-enabled by an explicit command to Kong's Admin API."